### ISSUE TITLE
Prevent multiple spaces before assignment operator during translation

### ISSUE THREAD

**Issue by itssamuelrowe** [pallene-lang/pallene#314]
Consider the following statement:
```
local x : integer = 10
```

The translator would output:

```
local x  = 10
```

Right now, the translator adds a synthetic space character if there is a space character before the assignment operator in the original input. However, if there is a space before the colon, it is not removed. This results in multiple spaces before the assignment operator after translation. The translator must be updated to prevent this behavior.

**Comment by hugomg** [comment]
I also noticed a couple of other things that look weird. We should try to come up with a solution that also makes them look nice as well.

First there is the case of the `as` operator leaving a space before it. In this example we end up with a trailing space before the end of the line:

```
local c = o[1] as string
```
```
local c = o[1] 
```

----------------

Another one is that for some reason the indentation after this comment is being removed:

```
export function binsearch(t: {integer}, x: integer) : integer
    -- lo <= x <= hi
    local lo = 1
    local hi = #t
```

```
function binsearch(t, x) 
    -- lo <= x <= hi
local lo = 1
    local hi = #t
```

**Comment by hugomg** [comment]
Fixed in #316
